lib/matplotlib/tests/test_colors.py

```diff
@@ -248,6 +248,17 @@ def test_LinearSegmentedColormap_from_list_bad_under_over():
     assert mcolors.same_color(cmap.get_over(), "y")
 
 
+def test_colormap_with_alpha():
+    cmap = ListedColormap(["red", "green", ("blue", 0.8)])
+    cmap2 = cmap.with_alpha(0.5)
+    # color is the same:
+    vals = [0, 0.5, 1]  # numeric positions that map to the listed colors
+    assert_array_equal(cmap(vals)[:, :2], cmap2(vals)[:, :2])
```

Comment:
There are 3 colour elements in RGBA:
```suggestion
    assert_array_equal(cmap(vals)[:, :3], cmap2(vals)[:, :3])
```